tests/testthat/test-Rules-methods.R

```diff
@@ -1043,3 +1043,91 @@ test_that("StoppingLowestDoseHSRBeta works correctly if first active dose is not
   )
   expect_identical(result, expected) # First active dose not applied.
 })
+
+# stopTrial-StoppingSpecificDose ----
+
+test_that("StoppingSpecificDose works correctly if dose rec. differs from specific", {
+  # StoppingSpecificDose works correctly if dose recommendation is not the same
+  # as the specific dose and stop is not met.
+  my_samples <- h_as_samples(
+    list(alpha0 = c(1.2, 0, -0.4, -0.1, 0.9), alpha1 = c(0.7, 1.7, 1.9, 0.6, 2.8))
+  )
+  result <- stopTrial(
+    stopping = h_stopping_specific_dose(),
+    dose = 20,
+    samples = my_samples,
+    model = h_get_logistic_log_normal(),
+    data = h_get_data_sr_1()
+  )
+  expected <- structure(
+    FALSE,
+    message = "Probability for target toxicity is 0 % for dose 80 and thus below the required 80 %"
+  )
+  expect_identical(result, expected)
+})
+
+test_that("StoppingSpecificDose works correctly if dose rec. differs from specific", {
```

Comment:
I don't understand how you want me to proceed .
Feel free to update any tests as you feel appropriate so that we can close this branch.
I have created 4 different test that test 2 x 2 different combinations 
stopping rule TRUE/FALSE x specific dose  the same = 80 /different =20 with the recommended dose.
I believe these cover both 1 and 2 of your points.

Reply:
I think just names of tests need to be updated?

Reply:
> I think just names of tests need to be updated?

Yes, exactly.

Reply:
OK I have updated/expanded the name of the tests. Hope this is fine now.

Reply:
Regarding the 2nd point , the function always returns the corrected message that always includes the specific dose in the message. Therefore when testing the results we are testing for the fact that the specific dose is indeed used and reported. Coming from that point I don't understand why we need extra tests for your 2nd point?

Reply:
> OK I have updated/expanded the name of the tests. Hope this is fine now.

Thank you. But the main point of my review is still not addressed.....

Reply:
What is the main point of your review? 
If this is about the your second point I have already expressed my throughs above that these test include also your 2nd point.

Reply:
> Regarding the 2nd point , the function always returns the corrected message that always includes the specific dose in the message. Therefore when testing the results we are testing for the fact that the specific dose is indeed used and reported. Coming from that point I don't understand why we need extra tests for your 2nd point?

Your function `stopTrial` can change the text of the message. The point here is that we do want to test that this functionally works as expected. So you need to create some unit test (or modify an existing ones) so that we see the effect of `gsub` function you used to change the text.

Reply:
> What is the main point of your review? If this is about the your second point I have already expressed my throughs above that these test include also your 2nd point.

No it does not, unfortunately.

Reply:
> > Regarding the 2nd point , the function always returns the corrected message that always includes the specific dose in the message. Therefore when testing the results we are testing for the fact that the specific dose is indeed used and reported. Coming from that point I don't understand why we need extra tests for your 2nd point?
> 
> Your function `stopTrial` can change the text of the message. The point here is that we do want to test that this functionally works as expected. So you need to create some unit test (or modify an existing ones) so that we see the effect of `gsub` function you used to change the text.

OK I understand but I am not sure I can do this on my own. So its better to get some help in order to save time and review comments.
Could you provide the modifications needed on the first test and then I can use this to update the rest?

Reply:
at this point it would be more effective to have short call and solve it together :-)

Reply:
Thanks @danielinteractive, sure - I am available anytime, @dimitriskontosClinBAY - please ping me on Slack when you have some time.